db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2200 @@
+                "00:00:00.000001",
+                "{[\"2024-01-01 00:02:00+00\",\"2024-01-01 00:04:00+00\"),[\"2024-01-01 00:07:00+00\",\"2024-01-01 00:09:00+00\")}",
+                "2024-01-04 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "E_event",
+                SOURCE_TYPE, "E",
+                DERIVATION_GROUP, "2024-01-01 00:13:00+00",
+                "00:00:00.000001",
+                "{[\"2024-01-01 00:13:00+00\",\"2024-01-01 00:20:00+00\")}",
+                "2024-01-05 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "F_event",
+                SOURCE_TYPE, "F",
+                DERIVATION_GROUP, "2024-01-01 00:04:00+00",
+                "00:00:00.000001",
+                "{[\"2024-01-01 00:04:00+00\",\"2024-01-01 00:07:00+00\")}",
+                "2024-01-06 00:00:00+00"
+            ),
+            new DerivedEvent(
+                "G_event",
+                SOURCE_TYPE, "G",
+                DERIVATION_GROUP, "2024-01-01 00:11:00+00",
+                "00:00:00.000001",
+                "{[\"2024-01-01 00:11:00+00\",\"2024-01-01 00:12:00+00\")}",
+                "2024-01-07 00:00:00+00"
+            )
+        );
+        assertEquals(expectedResults.size(), results.size());
+        assertTrue(results.containsAll(expectedResults));
+      }
+    }
+
+    /**
+     * This class focuses on testing each of the four rules involved in the derivation operation. To do so, new sources
+     *    and events are created, and their presence and intervals are tested to ensure correctness.
+     * The rules are as follows. Refer to the docs for more details on why they are what they are:
+     *    1. An External Event superceded by nothing will be present in the final, derived result.
+     *    2. An External Event partially superceded by a later External Source, but whose start time occurs before
+     *          the start of said External Source(s), will be present in the final, derived result.
+     *    3. An External Event whose start is superseded by another External Source, even if its end occurs after the
+     *          end of said External Source, will be replaced by the contents of that External Source (whether they are
+     *          blank spaces, or other events).
+     *    4. An External Event who shares a key with an External Event in a later External Source will always be
+     *          replaced.
+     */
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class DerivedEventRuleTests {
+
+      // Commonly Repeated:
+      @BeforeEach
+      void beforeEach() throws SQLException {
+        // insert generic external event type, source type, and derivation group
+        insertStandardTypes();
+      }
+
+      ////////////////////////// RULE 1 //////////////////////////
+      /**
+       * This test examines the lack of supercession by testing a solitary event. It vacuously shouldn't get superceded.
```

Comment:
Heads up: there are some more supercede-style typos in the comments. Also supercession isn’t a word. A simpler way to write this comment is:

```suggestion
       * A solitary event shouldn't be superseded by anything.
```